docs/proposals/admission_control_changes.md

```diff
@@ -0,0 +1,485 @@
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Admission control changes
+
+Admission control is the key security and cluster policy enforcement mechanism
+in Kubernetes. While in the future, we may wish to leverage additional
+mechanisms beyond the admission control point, given its depth of use by Kube
+users today, we should take steps to organize and streamline the existing
+functionality to better enable extension.
+
+We are proposing a series of changes to admission control in an attempt to make
+it less error-prone, easier to configure, and more flexible for admission plugin
+developers.
+
+
+## Hard-coded plugin execution order
+
+It's unlikely that administrators need to change the order in which admission
+plugins are executed. Furthermore, running certain plugins "out of order" will
+result in undesirable or incorrect behavior. For example, if you run
+`ResourceQuota` before `LimitRanger`, you could end up incorrectly calculating
+quota because it's possible for `LimitRanger` to modify pod values after
+`ResourceQuota` performed its quota checks.
+
+Because of this, we want to move to a model where in-tree admission plugin
+execution is strongly ordered and not under the administrator's control. We
+would therefore deprecate the `--admission-control` setting from
+`pkg/genericapiserver/options.ServerRunOptions`. During the deprecation period,
+if this flag is set, we will display a warning while continuing to honor the values
+specified by the flag (i.e., the administrator can override the hard-coded ordering).
+
+
+## One config file per plugin
+
+All admission plugins currently share a single config file, specified via
+`--admission-control-config-file`. This is problematic if multiple plugins
+use the same names for their configuration sections. For
+example, if plugin `Foo` wants
+
+```yaml
+myConfig:
+  someField: red
+```
+
+and plugin `Bar` wants
+
+```yaml
+myConfig:
+  someField:
+  - one
+  - two
+  - three
+```
+
+this won't work because of the naming collision and the type of `someField` is
+different for the two plugins. What we'd like to see instead is:
+
+1. deprecate `--admission-control-config-file`
+1. replace it with `--admission-plugin PLUGIN=CONFIG_FILE` (or an equivalent
```

Comment:
How would we plumb the appropriate configuration to an admission plugin when it's being initialized?

+1 to stopping adding flags. The only reason I am suggesting adding a new one here is because we've yet to realize moving all components from flags to config files/maps.